### DISCUSSION TITLE
Proposal: switch to libheif

### DISCUSSION THREAD

**Discussion by manfromarce** [tannerhelland/PhotoDemon#442] [Ideas]
[Libheif](https://github.com/strukturag/libheif) is a popular library which supports both AVIF and HEIF images.
Although PhotoDemon already supports these formats, switching to this library would provide the following benefits:
- The HEIF / HEVC add-on from Microsoft Store would no longer be required
- PhotoDemon could also encode HEIF images in addition to decoding them
- AVIF support wouldn't require an additional download (and maybe it would work on 32 bit too?) 

Replacing the currently used libraries could take a lot of work and I understand if it's not a priority, I'm just suggesting it but maybe this could also simplify PhotoDemon's code, since it should no longer check the availability of HEIF and AVIF plugins.

**Comment by tannerhelland** [comment]
Hello @manfromarce .  Thank you for this feedback.

I originally investigated libheif for use in PhotoDemon, but there are some complexities that make it a challenging choice.  The biggest issue at present is that the project does not provide any pre-built Windows libraries.  This means I need to maintain my own special library builds for PhotoDemon which is a large and time-consuming project.  (Other apps that use libheif typically have such support added by outside contributors - for example, the GIMP libheif plugin is provided by the authors of libheif themselves!  I can only dream of such contributions!)

Worse still, libheif only natively supports AVIF support in 64-bit builds.  This is not their fault; the underlying AVIF libraries they use (rav1e and dav1d) only support 64-bit Windows by default.  At present, I do not have the time or expertise to patch this support manually.

I could use libheif for HEIF images only, but again it is complex because I would need to maintain my own 32-bit library builds to work with PhotoDemon.  I have successfully built libheif in 32-bit mode and may look at adding it as a built-in part of PhotoDemon, but it is quite a large library and would increase download size significantly.  This is always the trade-off with supporting complex modern image formats.

I am currently working on JPEG-XL support, but will take another look at libheif in the coming weeks to see if any of these criteria have changed since I last investigated.  Thank you again for suggesting this.

**Comment by manfromarce** [comment]
Hi @tannerhelland , for my understanding [this project ](https://github.com/pphh77/libheif-Windowsbinary/releases) provides binaries for 64 bit Windows. Would it be userful for PhotoDemon? Once decompressed it is about 153 MB. Maybe instead of bundling it, Photodemon could offer to download it when the user tries to open an avif or heif image on 64 bit systems (like it currently does for the avif decoder), and keep using the Windows Heif add-on on 32 bit systems as fallback.

**Reply by wqweto** [reply]
> Hi @tannerhelland , for my understanding [this project ](https://github.com/pphh77/libheif-Windowsbinary/releases) provides binaries for 64 bit Windows. Would it be userful for PhotoDemon?

Tanner already said "I have successfully built libheif in 32-bit mode". Why would he want some other project's x64 build?

**Reply by manfromarce** [reply]
He said that it wouldn't include avif support and mantaining a special build would be a time-consuming task, so I thought this could help

**Reply by wqweto** [reply]
Once again, do you understand what "have successfully built libheif in 32-bit mode" means? He already has libheif built in 32-bit mode, ready to be used in the programming language he is using. He cannot use x64 builds of libheif in the programming language he is using.
